src/models/jwtPermissions/index.js

```diff
@@ -136,3 +155,74 @@ exports.init = async function ({ publicKeyUrl = _publicKeyUrl, loadPublicKeyFrom
         }
     }, 60 * 60 * 1000);
 };
+
+/**
+ *
+ * @param {string} tokenMerchantId
+ * @param {string} uriMerchantId
+ * @returns {void}
+ */
+const validateMerchant = (tokenMerchantId, uriMerchantId) => {
+    if (tokenMerchantId !== uriMerchantId && tokenMerchantId !== '*') {
+        throw new UserNotAuthorised(uriMerchantId);
+    }
+};
+/**
+ *
+ * @param {string[]} tokenPlaceIds
+ * @param {string|null} placeId
+ */
+const validatePlace = (tokenPlaceIds, placeId) => {
+    const validPlaceId = tokenPlaceIds.find((i) => i === placeId || i === '*');
+    if (!validPlaceId) {
+        throw new NotAuthorizedForPlace(placeId);
+    }
+
+};
+/**
+ *
+ * @param ctx
+ * @param next
+ * @returns {Promise<void>}
+ */
+exports.validateMerchantMiddleware = async (ctx, next) => {
+    const { jwtPayload } = ctx.state;
+    const permissionScope = getScope(jwtPayload, PERMISSION_SCOPE);
+    const { merchantId, placesIds } = permissionScope[2];
+    let uriMerchantId;
+    let uriPlaceId;
+    if (ctx.params.merchantPlaceId) {
+        [uriMerchantId, uriPlaceId] = ctx.params.merchantPlaceId.split('-');
+    } else {
+        uriMerchantId = ctx.params.merchantId;
+        uriPlaceId = ctx.params.placeId;
+    }
+    if (uriPlaceId) {
+        validatePlace(placesIds, uriPlaceId);
+    }
+    validateMerchant(merchantId, uriMerchantId);
+    await next();
+};
+
+/**
+ *
+ * @param {string} token
+ * @param {string} merchantId
+ * @param {number[]|number} permissions
+ * @param {string|undefined} publicKeyUrl
+ * @param {string|null} placeId
+ * @param {string|null} placeId
+ * @param {number[]|number|undefined} permissions
+ * @returns {Promise<void>}
+ */
+exports.authorizeUser = async function (
+    token, merchantId, { publicKeyUrl = _publicKeyUrl, placeId = null, permissions } = {}
+) {
+    const payload = await validateJwt(token, publicKeyUrl);
+    const permissionScope = getScope(payload, PERMISSION_SCOPE);
+    const tokenPermissions = permissionScope[1];
+    const { merchantId: tokenMerchantId, placesIds } = permissionScope[2];
```

Comment:
if the token does not contain the 'perms' scope, the code will fail

Reply:
done